org.d3cod3.Mosaic.yml

```diff
@@ -0,0 +1,449 @@
+app-id: org.d3cod3.Mosaic
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: mosaic
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
+  
+  - --socket=pulseaudio
+  # Needs access to webcams and so
+  - --device=dri
+  - --device=all
+  # Needs to talk to the network:
+  - --share=network
+  # Needs to save files locally
+  - --filesystem=host
+# cleanup:
+#  - /include
+add-extensions:
+    org.freedesktop.Platform.ffmpeg-full:
+      directory: lib/ffmpeg
+      version: '20.08'
+      add-ld-path: .
+      no-autodownload: true
+      autodelete: false
+cleanup-commands:
+  - mkdir -p ${FLATPAK_DEST}/lib/ffmpeg
+  - mkdir -p ${FLATPAK_DEST}/mosaic/apps/d3cod3/Mosaic/bin/data/tmp
+modules:
+# openFrameworks dependencies
+  - shared-modules/glu/glu-9.json
+  - shared-modules/glew/glew.json
+  - shared-modules/gtk2/gtk2.json
+  - shared-modules/libusb/libusb.json
+  - shared-modules/linux-audio/jack2.json
```

Comment:
To support JACK you need to add the following `finish-args`
`--filesystem=xdg-run/pipewire-0`

This will route JACK through PipeWire.

Reply:
and the jack libraries should also be deleted so that the pipewire provided one be used.

Reply:
I'm curious about this one. I knew pipewire natively routed JACK and ALSA, but not how to do it in flatpak. Does it need pipewire on host? Fedora users would be fine, but I'm not that optimistic about ubuntu ones

Reply:
it works by:
- having the above finish arg to access pipewire
- removing the jack `.so` so that the pipewire provide be open (they are in the 20.08 runtime).

pipewire need to be installed on the host, and there is no alternative. 

Otherwise disable JACK altogether.

Reply:
allowing access to the system bug *might* be needed, but it's on a case by case basis.

Reply:
Another doubt: does --filesystem=host superseed --filesystem=xdg-run/pipewire-0, or do I need both?

Reply:
Ideally it should be `--filesystem=home` instead of `host` but the other one is necessary.